### PR TITLE
Study programme & curriculum to whoami

### LINKED ISSUE #5938: study program information to whoami endpoint

### PR DESCRIPTION
**Pull request by saaralepisto**
Closes #5938

### CODE REVIEW

File: muikku-rest/src/main/java/fi/otavanopisto/muikku/rest/user/UserRESTService.java [hunk 1]
```diff
@@ -1647,6 +1652,15 @@ public Response findWhoAmI(@Context Request request) {
     OffsetDateTime studyTimeEnd = user == null ? null : user.getStudyTimeEnd();
     String studyTimeLeftStr = userEntityController.getStudyTimeEndAsString(studyTimeEnd);
     
+    // Curriculum 
+    String curriculumName = null;
+    if (user.getCurriculumIdentifier() != null) {
```
Comment (pkukkon) [inline muikku-rest/src/main/java/fi/otavanopisto/muikku/rest/user/UserRESTService.java:1657]: If user is null, the code will crash here.

File: muikku-rest/src/main/java/fi/otavanopisto/muikku/rest/user/UserRESTService.java [hunk 2]
```diff
@@ -1647,6 +1652,15 @@ public Response findWhoAmI(@Context Request request) {
...
+    SchoolDataIdentifier curriculumId = SchoolDataIdentifier.fromId(user.getCurriculumIdentifier());
```
Comment (pkukkon) [inline muikku-rest/src/main/java/fi/otavanopisto/muikku/rest/user/UserRESTService.java:1658]: Proper indentation

File: muikku-rest/src/main/java/fi/otavanopisto/muikku/rest/user/UserRESTService.java [hunk 3]
```diff
@@ -1647,6 +1652,15 @@ public Response findWhoAmI(@Context Request request) {
...
+    
+    Curriculum curriculum = user == null ? null : courseMetaController.findCurriculum(curriculumId.getDataSource(), curriculumId.getIdentifier());
```
Comment (pkukkon) [inline muikku-rest/src/main/java/fi/otavanopisto/muikku/rest/user/UserRESTService.java:1660]: user cannot be null at this point; the code would already have crashed before.

File: muikku-rest/src/main/java/fi/otavanopisto/muikku/rest/user/UserRESTService.java [hunk 4]
```diff
@@ -1647,6 +1652,15 @@ public Response findWhoAmI(@Context Request request) {
...
+    
+    curriculumName = curriculum.getName();
```
Comment (pkukkon) [inline muikku-rest/src/main/java/fi/otavanopisto/muikku/rest/user/UserRESTService.java:1662]: If findCurriculum would return null, the code would crash at this point.

**Review by pkukkon — CHANGES_REQUESTED** [review]

**Review by pkukkon — APPROVED** [review]

**Review by jjlankinen — APPROVED** [review]
This has been tested locally on 3 different occasions and has been approved earlier.

**Pull request merged**